src/Middleware/Spa/SpaProxy/src/SpaProxyStartupFilter.cs

```diff
@@ -3,13 +3,32 @@
 
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
 
 namespace Microsoft.AspNetCore.SpaProxy;
 
 internal sealed class SpaProxyStartupFilter : IStartupFilter
 {
+    private readonly SpaProxyLaunchManager _spaProxyLaunchManager;
+    private readonly IHostApplicationLifetime _hostLifetime;
+
+    public SpaProxyStartupFilter(
+        SpaProxyLaunchManager spaProxyLaunchManager,
+        IHostApplicationLifetime hostLifetime)
+    {
+        _spaProxyLaunchManager = spaProxyLaunchManager ?? throw new ArgumentNullException(nameof(spaProxyLaunchManager));
+        _hostLifetime = hostLifetime ?? throw new ArgumentNullException(nameof(hostLifetime));
+    }
+
     public Action<IApplicationBuilder> Configure(Action<IApplicationBuilder> next)
     {
+        Task.Run(async () =>
+        {
+            if (!await _spaProxyLaunchManager.IsSpaProxyRunning(_hostLifetime.ApplicationStopping))
+            {
+                _spaProxyLaunchManager.StartInBackground(_hostLifetime.ApplicationStopping);
+            }
```

Comment:
This needs to be inside a try..catch since we are not observing the result of the task, unless `IsSpaProxyRunning` and `StartInBackground` are guaranteed not to fail.